test/user.test.ts

```diff
@@ -0,0 +1,106 @@
+import * as request from 'supertest';
+import { expect } from 'chai';
+import { User } from '../src/entity/user';
+import { generateJwt } from '../src/token';
+
+describe('Query user', () => {
+  beforeEach(async () => {
+    const firstUser = new User();
+    const secondUser = new User();
+
+    firstUser.create({
+      name: 'First User',
+      email: 'first@user.com',
+      password: 'some_password',
+      birthDate: '01-01-2001',
+    });
+
+    secondUser.create({
+      name: 'Second User',
+      email: 'second@user.com',
+      password: 'some_password',
+      birthDate: '01-01-2001',
+    });
+
+    await User.getRepository().save(firstUser);
+    await User.getRepository().save(secondUser);
+  });
+
+  afterEach(async () => {
+    queryUser.variables.id = 0;
+    const userRepository = User.getRepository();
+    await userRepository.clear();
+  });
+
+  it('should run query and return user info', async () => {
+    const userRepository = User.getRepository();
+    const firstUser = await userRepository.findOne({ email: 'first@user.com' });
+    const secondUser = await userRepository.findOne({ email: 'second@user.com' });
+
+    const firstUserToken = generateJwt(firstUser, false);
+    queryUser.variables.id = secondUser.id;
+
+    const response = await request('localhost:4000').post('/').send(queryUser).set('Authorization', firstUserToken);
+    const responseUserData = response.body.data.user;
+
+    expect(responseUserData.id).to.be.equal(secondUser.id);
+    expect(responseUserData.name).to.be.equal(secondUser.name);
+    expect(responseUserData.email).to.be.equal(secondUser.email);
+    expect(responseUserData.birthDate).to.be.equal(secondUser.birthDate);
```

Comment:
Pros objetos retornados pelo GraphQL eu sugiro vc usar `expect(...).to.be.deep.eq(...)`, pq ele faz uam comparação estrita de objetos e deixa o teste mais preciso.